x-pack/plugins/elastic_assistant/server/lib/langchain/graphs/default_assistant_graph/index.ts

```diff
@@ -114,7 +112,7 @@ export const callAssistantGraph: AgentExecutor<true | false> = async ({
     isEnabledKnowledgeBase,
     kbDataClient: dataClients?.kbDataClient,
     logger,
-    modelExists,
+    modelExists: isEnabledKnowledgeBase,
```

Comment:
I think we can merge the `isEnabledKnowledgeBase` and `modelExists` `AssistantToolParams` into a single param as they both mean the same thing right now. If we need a distinction between kbSetup and if ESQL docs have been loaded we can use the new `kbDataClient.isESQLDocsLoaded()` helper since all tools have access to the `kbDataClient`.